engine/src/main/java/org/terasology/rendering/dag/nodes/ShadowMapNode.java

```diff
@@ -72,86 +81,113 @@
     @In
     private ShadowMapResolutionDependentFBOs shadowMapResolutionDependentFBOs;
 
-    private Material shadowMapShader;
     private RenderingConfig renderingConfig;
     private Camera playerCamera;
     private float texelSize;
-    private float stepSize;
-
 
     @Override
     public void initialise() {
         this.playerCamera = worldRenderer.getActiveCamera();
-        this.shadowMapShader = worldRenderer.getMaterial("engine:prog.shadowMap");
         this.renderingConfig = config.getRendering();
         renderableWorld.setShadowMapCamera(shadowMapCamera);
 
         requiresFBO(new FBOConfig(SHADOW_MAP, FBO.Type.NO_COLOR).useDepthBuffer(), shadowMapResolutionDependentFBOs);
 
-        // TODO: fix them
+        texelSize = 1.0f / renderingConfig.getShadowMapResolution() * 2.0f;
+        renderingConfig.subscribe(RenderingConfig.SHADOW_MAP_RESOLUTION, this);
+
+        requiresCondition(() -> renderingConfig.isDynamicShadows());
+        renderingConfig.subscribe(RenderingConfig.DYNAMIC_SHADOWS, this);
```

Comment:
We might probably need an helper function for these lines, I think it started to get a little redundant especially for the nodes that are `ConditionDependentNode`. But I am not sure about how to implement them.